src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -1364,6 +1364,25 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 		}
 	}
 
+	async updateAll(): Promise<InstallExtensionResult[]> {
+		const toUpdate: InstallExtensionInfo[] = [];
+		this.outdated.forEach((extension) => {
+			if (extension.gallery) {
+				toUpdate.push({
+					extension: extension.gallery,
+					options: {
+						operation: InstallOperation.Update,
+						installPreReleaseVersion: extension.local?.isPreReleaseVersion,
+						profileLocation: this.userDataProfileService.currentProfile.extensionsResource,
+						donotIncludePackAndDependencies: true,
```

Comment:
@sandy081 I found that without `donotIncludePackAndDependencies: true` an "updateAll" of deliberately downdated Python + Pylance extensions would result in unterminated progress indicator and the "Reload Required" buttons never appeared. My hunch is the dependency that Python has on Pylance is the cause, If it's not acceptable to use this option here I'd like your help working out why the `installGalleryExtensions` promises don't all resolve in the Python + Pylance update example.

Reply:
I will look into this. May be a bug in bulk install method.

Reply:
Yes, this is a bug in the bulk install method, I will fix it separately. 

Recommended to go with this flag and file an issue for this bug (for tracking purpose).

Reply:
This is the bug we are hitting here - https://github.com/microsoft/vscode/issues/195690

Reply:
Fixed the bug. You can remove this line